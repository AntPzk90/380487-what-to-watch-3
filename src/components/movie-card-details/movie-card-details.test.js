import React from 'react';
import renderer from 'react-test-renderer';
import MovieCardDetails from './../movie-card-details/movie-card-details.jsx';
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

const filmMock = {
  title: `Dardjeeling limited`,
  src: `dardjeeling-limited.jpg`,
  poster: `dardjeeling-limited.jpg`,
  titlePoster: `dardjeeling-limited.jpg`,
  genre: `Comedy`,
  releaseDate: `2016`,
};

it(`SnapshotTest MovieCardDetails`, () => {

  const store = mockStore({
    films: filmMock,
  });

  const tree = renderer
  .create(
      <Provider store={store}>
        <MovieCardDetails
          film = {filmMock}
        />
      </Provider>, {
        createNodeMock: () => {
          return {};
        }
      })
  .toJSON();

  expect(tree).toMatchSnapshot();
});
