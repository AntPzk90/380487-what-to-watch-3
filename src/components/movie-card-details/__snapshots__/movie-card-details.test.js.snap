// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapshotTest MovieCardDetails 1`] = `
<div
  className="movie-card__text movie-card__row"
>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
      >
        Wes Andreson
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <span
        className="movie-card__details-value"
      >
        <span>
          Tom Cruise
           
          <br />
        </span>
        <span>
          Dakota Fanning
           
          <br />
        </span>
        <span>
          Tim Robbins
           
          <br />
        </span>
      </span>
    </p>
  </div>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run Time
      </strong>
      <span
        className="movie-card__details-value"
      >
        116
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
      >
        Adventure
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
      >
        2005
      </span>
    </p>
  </div>
</div>
`;
